### PR TITLE
relocate config import

### PR DESCRIPTION
**Pull request by cmazakas**
There's a vcpkg bug where it implicitly relies on certain aspects of file layout which prevents Buffers from being packaged.

Buffers is the only compiled Boost library where vcpkg breaks as well.

To make life easier for everyone involved and to bring uniformity to Buffers, we just relocate where config is imported by the Jamfiles.

### CODE REVIEW

File: Jamfile [hunk 1]
```diff
@@ -7,16 +7,4 @@
...
-    [ requires
-    cxx11_constexpr
-    cxx11_decltype
-    cxx11_hdr_tuple
-    cxx11_template_aliases
-    cxx11_variadic_templates
-    ]
-    ;
-
 build-project test ;
```
Comment (ashtum) [inline Jamfile:10]: Is this necessary?

Comment (cmazakas) [reply Jamfile:10]: That's actually a good question. I can try digging into this to see if this is genuinely necessary or not.

Comment (vinniefalco) [reply Jamfile:10]: I believe the point of that is so "b2" from the root will build tests. But I am not 100% sure.

Comment (cmazakas) [reply Jamfile:10]: Ah, so it is. I've tested locally. Without it, no tests run from the project root. Running `b2 libs/buffers` still works as expected. We can keep this. I don't mind now that we know the functionality.

Do you guys think I should update the root Jamfile to contain a comment summarizing this?

Comment (ashtum) [reply Jamfile:10]: > Without it, no tests run from the project root.

Do we need to be able to run tests from the root?

Comment (vinniefalco) [reply Jamfile:10]: Peter doesn't like a top-level Jamfile so we should get rid of it.

**Pull request merged**